dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/EnvConfig13Source.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config13.sources;
+
+import java.util.regex.Pattern;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.microprofile.config.interfaces.ConfigConstants;
+import com.ibm.ws.microprofile.config.sources.EnvConfigSource;
+
+/**
+ * Enhancements to the EnvConfigSource class for MP Config 1.3.
+ */
+public class EnvConfig13Source extends EnvConfigSource {
+
+    private static final TraceComponent tc = Tr.register(EnvConfig13Source.class);
+    private Pattern p = null;
+
+    public EnvConfig13Source() {
+        p = Pattern.compile(ConfigConstants.ALLOWABLE_CHARS_IN_ENV_VAR_SOURCE);
```

Comment:
I would have thought that this could be statically compiled, not in the constructor?

Reply:
I wonder if we should also override the name of this config source to say 1.3 (just like you have done for the toString())??

Reply:
(The spec isn't that clear on how unique the names should be)